frmts/tiledb/tiledbsparse.cpp

```diff
@@ -0,0 +1,4934 @@
+/* ==================================================================== */
+/*                            OGRTileDBDataset                          */
+/* ==================================================================== */
+/************************************************************************/
+
+/************************************************************************/
+/*                         OGRTileDBDataset()                           */
+/************************************************************************/
+
+OGRTileDBDataset::OGRTileDBDataset()
+{
+}
+
+/************************************************************************/
+/*                         ~OGRTileDBDataset()                          */
+/************************************************************************/
+
+OGRTileDBDataset::~OGRTileDBDataset()
+{
+}
+
+/************************************************************************/
+/*                                Open()                                */
+/************************************************************************/
+
+GDALDataset *OGRTileDBDataset::Open(GDALOpenInfo *poOpenInfo,
+                                    tiledb::Object::Type objectType)
+
+{
+    auto poDS = cpl::make_unique<OGRTileDBDataset>();
+    poDS->eAccess = poOpenInfo->eAccess;
+    const char *pszConfig =
+        CSLFetchNameValue(poOpenInfo->papszOpenOptions, "TILEDB_CONFIG");
+
+    const char *pszTimestamp = CSLFetchNameValueDef(
+        poOpenInfo->papszOpenOptions, "TILEDB_TIMESTAMP", "0");
+    const uint64_t nTimestamp = std::strtoull(pszTimestamp, nullptr, 10);
+
+    if (pszConfig != nullptr)
+    {
+        tiledb::Config cfg(pszConfig);
+        poDS->m_ctx.reset(new tiledb::Context(cfg));
+    }
+    else
+    {
+        tiledb::Config cfg;
+        cfg["sm.enable_signal_handlers"] = "false";
+        poDS->m_ctx.reset(new tiledb::Context(cfg));
+    }
+
+    std::string osFilename(
+        TileDBDataset::VSI_to_tiledb_uri(poOpenInfo->pszFilename));
+    if (osFilename.back() == '/')
+        osFilename.pop_back();
+
+    const auto AddLayer = [&poDS, nTimestamp, poOpenInfo](
+                              const std::string &osLayerFilename,
+                              const std::optional<std::string> &osLayerName =
+                                  std::optional<std::string>())
+    {
+        auto poLayer = cpl::make_unique<OGRTileDBLayer>(
+            osLayerFilename.c_str(),
+            osLayerName.has_value() ? (*osLayerName).c_str()
+                                    : CPLGetBasename(osLayerFilename.c_str()),
+            wkbUnknown, nullptr);
+        poLayer->m_bUpdatable = poOpenInfo->eAccess == GA_Update;
+        if (!poLayer->InitFromStorage(poDS->m_ctx.get(), nTimestamp,
+                                      poOpenInfo->papszOpenOptions))
+        {
+            return false;
+        }
+
+        poLayer->m_nBatchSize =
```

Comment:
`batch_size` should be larger, `ogr2ogr -f TileDB nz_tiledb nz-building-outlines.shp -lco BATCH_SIZE=500000` gives fragment sizes that are around 150MB, it is dataset dependent but the batch size above seems reasonable.

Reply:
done